[PLATFORM-325]: [BridgeEx] Give some love to the library

cpiemontese opened the pull request:
https://prima-assicurazioni-spa.myjetbrains.com/youtrack/issue/PLATFORM-325

This PR tries to improve and give some love to bridge_ex

- add some module documentation
- improve test coverage
- better handling of local vs global configs (`log_options` can now be configured globally too)
- general code refactor

pdincau reviewed (commented):
@cpiemontese do we need a changelog update?

cpiemontese commented:
> @cpiemontese do we need a changelog update?

I updated the changelog, but I am unsure about the version.

We moved an option from being local to global. Technically the functionality is still there and the only thing that changes is how it's configured, but it may still be considered breaking...?

cottinisimone reviewed (dismissed):
Bel lavoro

cpiemontese commented:
I fixed the CHANGELOG as we discussed i.e. instead of specifying a version everything goes under next. I think this can now be safely merged

fabuloso reviewed (approved):
ggwp
